src/commands/bundle_for_gem.yml

```diff
@@ -8,10 +8,17 @@ parameters:
   project:
     type: string
 steps:
+  # md5sum's Gemfile and << parameters.project >>.gemspec (optionally, if it exists) to generate a unique
+  # cache key representing the contents of both files.
+  - run:
+      name: Generate a cache key that optionally includes a .gemspec file
+      command: |
+        echo $(find . -d 1 -type f \( -name Gemfile -o -name << parameters.project >>.gemspec \) -exec md5sum {} \; | md5sum | cut -d' ' -f1) >> BUNDLE_CACHE_KEY
```

Comment:
For Avalon, we commit our Gemfile.lock so maybe that should be included in this cache key computation since it will be more accurate than the Gemfile?